HOApp/ViewModel/POEntryViewModel.cs

```diff
@@ -1,106 +1,129 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
-using System.Data.Entity;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.ComponentModel;
 using System.Windows;
 using DataAccess;
-using DataAccess.Entity.Entities;
-using DataAccess.Help;
 using HOApp.Model;
+using System.Data.Entity;
 
 namespace HOApp.ViewModel
 {
-    class POEntryViewModel : DataVMBase
+    internal class POEntryViewModel : DataVMBase
     {
-        public RetailDbContext db = new RetailDbContext();
+        public POEntryViewModel() : base()
+        {
+            _selectableProducts = new List<ProductVM>();
+            PO = new POVM();
+            PropertyChanged += HandlePropertyChanged;
+        }
+
+        private void HandlePropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            switch (e.PropertyName)
+            {
+                case nameof(SelectedSupplier):
+                    {
+                        var suppliersId = _selectedSupplier?.TheEntity.SupplierID ?? -1;
+
+                        using (var db = new RetailDbContext())
+                        {
+                            _selectableProducts = db
+                                .Products
+                                .Where(product => product.SupplierID == suppliersId)
+                                .OrderBy(n => n.Description)
+                                .Select(n => new ProductVM { IsNew = false, TheEntity = n })
+                                .ToList();
+                        }

```

Comment:
When supplier changes, changes the selectable products.

Reply:
Great :) I was going to do this. Product.supplierId is just the primary supplier. By having a new junction table 'productsupplier', a product can have multiple suppliers.